File: API/demo/src/main/java/com/example/demo/model/PhysicalActivity.java
package com.example.demo.model;

import com.example.demo.Enum.PhysicalActivityTipe;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;

import java.time.LocalDate;
import java.util.Date;

@Entity
public class PhysicalActivity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Users usuario;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate fecha;
    @Enumerated(EnumType.STRING)
    private PhysicalActivityTipe tipo;
    private int duracion;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Users getUsuario() {
        return usuario;
    }

    public void setUsuario(Users usuario) {
        this.usuario = usuario;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public PhysicalActivityTipe getTipo() {
        return tipo;
    }

    public void setTipo(PhysicalActivityTipe tipo) {
        this.tipo = tipo;
    }

    public int getDuracion() {
        return duracion;
    }

    public void setDuracion(int duracion) {
        this.duracion = duracion;
    }
}

File: Aplicacion Movil/MedMonitor/app/src/main/java/Models/PhysicalActivity.kt
package Models

import Enums.PhysicalActivityTipe
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.time.LocalDate

data class PhysicalActivity(
    @SerializedName("id") val id: Long,
    @SerializedName("usuario_id") val usuarioId: Int,
    @SerializedName("fecha") val fecha: LocalDate,
    @SerializedName("tipo") val tipo: PhysicalActivityTipe,
    @SerializedName("duracion") val duracion: Int
):Serializable
